packages/bson-transpilers/symbols/java/templates.yaml

```diff
@@ -903,7 +903,7 @@ Templates:
     ObjectIdCreateFromTimeArgsTemplate: &ObjectIdCreateFromTimeArgsTemplate !!js/function >
         (lhs, arg, isNumber) => {
             if (isNumber) {
-                return `(new java.util.Date(${arg}))`;
+                return `(new java.util.Date(${arg.replace(/L$/, '000L')}))`;
```

Comment:
Java dates take ms arguments (and since we specify this as a long, we cannot just multiply)